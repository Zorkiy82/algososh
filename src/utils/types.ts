export type TSwapLog = [number, number][];