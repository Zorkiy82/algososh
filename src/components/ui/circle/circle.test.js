import { render, screen, fireEvent } from "@testing-library/react";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe("Тесты компонента Circle", () => {
  it("Кейс #1 (без буквы и вообще без пропов) корректно отрисовывается", () => {
    render(<Circle />);
    const element = screen.getByTestId("element-id");
    const letter = screen.getByTestId("letter-id");
    const head = screen.getByTestId("head-id");
    const tail = screen.getByTestId("tail-id");
    const index = screen.getByTestId("index-id");
    const circle = screen.getByTestId("circle-id");
    expect(letter).toHaveTextContent("");
    expect(head).toHaveTextContent("");
    expect(tail).toHaveTextContent("");
    expect(index).toHaveTextContent("");
    expect(circle.classList.contains("small")).toBe(false);
    expect(circle.classList.contains(ElementStates.Default)).toBe(true);
    expect(circle.classList.contains(ElementStates.Modified)).toBe(false);
    expect(circle.classList.contains(ElementStates.Changing)).toBe(false);
    expect(element).toMatchSnapshot();
  });

  it("Кейс #2 (с буквами) корректно отрисовывается", () => {
    render(<Circle letter="aBcdD" />);
    const element = screen.getByTestId("element-id");
    const letter = screen.getByTestId("letter-id");
    expect(letter).toHaveTextContent(/aBcdD/i);
    expect(element).toMatchSnapshot();
  });

  it("Кейс #3 (с head) корректно отрисовывается", () => {
    render(<Circle head={"head"} />);
    const element = screen.getByTestId("element-id");
    const head = screen.getByTestId("head-id");
    expect(head).toHaveTextContent("head");
    expect(element).toMatchSnapshot();
  });

  it("Кейс #4 (с react-элементом в head) корректно отрисовывается", () => {
    const { rerender } = render(<Circle />);
    const element = screen.getByTestId("element-id");
    let heads = screen.queryAllByTestId("head-id");
    expect(heads).toHaveLength(1);
    expect(heads[0].childElementCount).toBe(0);
    expect(element).toMatchSnapshot();

    rerender(<Circle head={<Circle />} />);
    heads = screen.queryAllByTestId("head-id");
    expect(heads).toHaveLength(2);
    expect(heads[0].childElementCount).toBe(1);
    expect(element).toMatchSnapshot();
  });

  it("Кейс #5 (с tail) корректно отрисовывается", () => {
    render(<Circle tail={"tail"} />);
    const element = screen.getByTestId("element-id");
    const tail = screen.getByTestId("tail-id");
    expect(tail).toHaveTextContent("tail");
    expect(element).toMatchSnapshot();
  });

  it("Кейс #6 (с react-элементом в tail) корректно отрисовывается", () => {
    const { rerender } = render(<Circle />);
    const element = screen.getByTestId("element-id");
    let tails = screen.queryAllByTestId("tail-id");
    expect(tails).toHaveLength(1);
    expect(tails[0].childElementCount).toBe(0);
    expect(element).toMatchSnapshot();

    rerender(<Circle tail={<Circle />} />);
    tails = screen.queryAllByTestId("tail-id");
    expect(tails).toHaveLength(2);
    expect(tails[0].childElementCount).toBe(1);
    expect(element).toMatchSnapshot();
  });

  it("Кейс #7 (с index) корректно отрисовывается", () => {
    render(<Circle index={1} />);
    const element = screen.getByTestId("element-id");
    const index = screen.getByTestId("index-id");
    expect(index).toHaveTextContent("1");
    expect(element).toMatchSnapshot();
  });

  it("Кейс #8 (с пропом isSmall ===  true) корректно отрисовывается", () => {
    render(<Circle isSmall={true} />);
    const element = screen.getByTestId("element-id");
    const circle = screen.getByTestId("circle-id");
    expect(circle.classList.contains("small")).toBe(true);
    expect(element).toMatchSnapshot();
  });

  it("Кейс #9 (в состоянии default) корректно отрисовывается", () => {
    render(<Circle state={ElementStates.Default} />);
    const element = screen.getByTestId("element-id");
    const circle = screen.getByTestId("circle-id");
    expect(circle.classList.contains(ElementStates.Default)).toBe(true);
    expect(circle.classList.contains(ElementStates.Modified)).toBe(false);
    expect(circle.classList.contains(ElementStates.Changing)).toBe(false);
    expect(element).toMatchSnapshot();
  });

  it("Кейс #10 (в состоянии changing) корректно отрисовывается", () => {
    render(<Circle state={ElementStates.Changing} />);
    const element = screen.getByTestId("element-id");
    const circle = screen.getByTestId("circle-id");
    expect(circle.classList.contains(ElementStates.Changing)).toBe(true);
    expect(circle.classList.contains(ElementStates.Default)).toBe(false);
    expect(circle.classList.contains(ElementStates.Modified)).toBe(false);
    expect(element).toMatchSnapshot();
  });

  it("Кейс #11 (в состоянии modified) корректно отрисовывается", () => {
    render(<Circle state={ElementStates.Modified} />);
    const element = screen.getByTestId("element-id");
    const circle = screen.getByTestId("circle-id");
    expect(circle.classList.contains(ElementStates.Modified)).toBe(true);
    expect(circle.classList.contains(ElementStates.Changing)).toBe(false);
    expect(circle.classList.contains(ElementStates.Default)).toBe(false);
    expect(element).toMatchSnapshot();
  });


});
