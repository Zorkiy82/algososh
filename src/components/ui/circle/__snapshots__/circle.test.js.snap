// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тесты компонента Circle Кейс #1 (без буквы и вообще без пропов) корректно отрисовывается 1`] = `
<div
  class="content "
  data-testid="element-id"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-id"
  />
  <div
    class="circle   default"
    data-testid="circle-id"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="letter-id"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index-id"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail-id"
  />
</div>
`;

exports[`Тесты компонента Circle Кейс #2 (с буквами) корректно отрисовывается 1`] = `
<div
  class="content "
  data-testid="element-id"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-id"
  />
  <div
    class="circle   default"
    data-testid="circle-id"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="letter-id"
    >
      aBcdD
    </p>
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index-id"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail-id"
  />
</div>
`;

exports[`Тесты компонента Circle Кейс #3 (с head) корректно отрисовывается 1`] = `
<div
  class="content "
  data-testid="element-id"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head string"
    data-testid="head-id"
  >
    head
  </div>
  <div
    class="circle   default"
    data-testid="circle-id"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="letter-id"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index-id"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail-id"
  />
</div>
`;

exports[`Тесты компонента Circle Кейс #4 (с react-элементом в head) корректно отрисовывается 1`] = `
<div
  class="content "
  data-testid="element-id"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-id"
  />
  <div
    class="circle   default"
    data-testid="circle-id"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="letter-id"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index-id"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail-id"
  />
</div>
`;

exports[`Тесты компонента Circle Кейс #4 (с react-элементом в head) корректно отрисовывается 2`] = `
<div
  class="content "
  data-testid="element-id"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-id"
  >
    <div
      class="content "
      data-testid="element-id"
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="head-id"
      />
      <div
        class="circle   default"
        data-testid="circle-id"
      >
        <p
          class="text text_type_circle text_color_input letter"
          data-testid="letter-id"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
        data-testid="index-id"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="tail-id"
      />
    </div>
  </div>
  <div
    class="circle   default"
    data-testid="circle-id"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="letter-id"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index-id"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail-id"
  />
</div>
`;

exports[`Тесты компонента Circle Кейс #5 (с tail) корректно отрисовывается 1`] = `
<div
  class="content "
  data-testid="element-id"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-id"
  />
  <div
    class="circle   default"
    data-testid="circle-id"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="letter-id"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index-id"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-testid="tail-id"
  >
    tail
  </div>
</div>
`;

exports[`Тесты компонента Circle Кейс #6 (с react-элементом в tail) корректно отрисовывается 1`] = `
<div
  class="content "
  data-testid="element-id"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-id"
  />
  <div
    class="circle   default"
    data-testid="circle-id"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="letter-id"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index-id"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail-id"
  />
</div>
`;

exports[`Тесты компонента Circle Кейс #6 (с react-элементом в tail) корректно отрисовывается 2`] = `
<div
  class="content "
  data-testid="element-id"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-id"
  />
  <div
    class="circle   default"
    data-testid="circle-id"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="letter-id"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index-id"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail-id"
  >
    <div
      class="content "
      data-testid="element-id"
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="head-id"
      />
      <div
        class="circle   default"
        data-testid="circle-id"
      >
        <p
          class="text text_type_circle text_color_input letter"
          data-testid="letter-id"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
        data-testid="index-id"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="tail-id"
      />
    </div>
  </div>
</div>
`;

exports[`Тесты компонента Circle Кейс #7 (с index) корректно отрисовывается 1`] = `
<div
  class="content "
  data-testid="element-id"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-id"
  />
  <div
    class="circle   default"
    data-testid="circle-id"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="letter-id"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index-id"
  >
    1
  </p>
  <div
    class="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="tail-id"
  />
</div>
`;

exports[`Тесты компонента Circle Кейс #8 (с пропом isSmall ===  true) корректно отрисовывается 1`] = `
<div
  class="content "
  data-testid="element-id"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-id"
  />
  <div
    class="circle  small default"
    data-testid="circle-id"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="letter-id"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index-id"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail-id"
  />
</div>
`;

exports[`Тесты компонента Circle Кейс #9 (в состоянии default) корректно отрисовывается 1`] = `
<div
  class="content "
  data-testid="element-id"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-id"
  />
  <div
    class="circle   default"
    data-testid="circle-id"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="letter-id"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index-id"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail-id"
  />
</div>
`;

exports[`Тесты компонента Circle Кейс #10 (в состоянии changing) корректно отрисовывается 1`] = `
<div
  class="content "
  data-testid="element-id"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-id"
  />
  <div
    class="circle   changing"
    data-testid="circle-id"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="letter-id"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index-id"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail-id"
  />
</div>
`;

exports[`Тесты компонента Circle Кейс #11 (в состоянии modified) корректно отрисовывается 1`] = `
<div
  class="content "
  data-testid="element-id"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head-id"
  />
  <div
    class="circle   modified"
    data-testid="circle-id"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="letter-id"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index-id"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail-id"
  />
</div>
`;
